include/RMGManager.hh

```diff
@@ -51,62 +67,191 @@ class RMGManager {
       return fG4RunManager->GetRunManagerType() == G4RunManager::RMType::sequentialRM;
     }
 
     // setters
+    /**
+     * @brief Sets the Geant4 run manager.
+     * @param g4_manager Pointer to the G4RunManager.
+     */
     void SetUserInit(G4RunManager* g4_manager) {
       fG4RunManager = std::unique_ptr<G4RunManager>(g4_manager);
     }
+    /**
+     * @brief Sets the Geant4 visualization manager.
+     * @param vis Pointer to the G4VisManager.
+     */
     void SetUserInit(G4VisManager* vis) { fG4VisManager = std::unique_ptr<G4VisManager>(vis); }
+    /**
+     * @brief Sets the detector construction.
+     * @param det Pointer to RMGHardware.
+     */
     void SetUserInit(RMGHardware* det) { fDetectorConstruction = det; }
+    /**
+     * @brief Sets the physics list.
+     * @param proc Pointer to G4VUserPhysicsList.
+     */
     void SetUserInit(G4VUserPhysicsList* proc) { fPhysicsList = proc; }
 
+    /**
+     * @brief Enables or disables interactive mode.
+     * @param flag True to enable interactive mode.
+     */
     void SetInteractive(bool flag = true) { fInteractive = flag; }
+    /**
+     * @brief Sets the number of threads.
+     * @param nthreads Number of threads.
+     */
     void SetNumberOfThreads(int nthreads) { fNThreads = nthreads; }
+    /**
+     * @brief Sets the print modulo value.
+     * @param n_ev Number of events for modulo printing.
+     */
     void SetPrintModulo(int n_ev) { fPrintModulo = n_ev > 0 ? n_ev : -1; }
 
+    /**
+     * @brief Includes a macro file for execution.
+     * @param filename The name of the macro file.
+     */
     void IncludeMacroFile(std::string filename) { fMacroFilesOrContents.emplace_back(filename); }
+    /**
+     * @brief Registers a Geant4 alias.
+     * @param alias The alias name.
+     * @param value The corresponding value.
+     */
     void RegisterG4Alias(std::string alias, std::string value) { fG4Aliases.emplace(alias, value); }
+    /**
+     * @brief Initializes the simulation components.
+     */
     void Initialize();
+    /**
+     * @brief Executes the simulation run.
+     */
     void Run();
 
+    /**
+     * @brief Sets the random engine by name.
+     * @param name Name of the random engine.
+     */
     void SetRandEngine(std::string name);
+    /**
+     * @brief Sets the seed for the random engine.
+     * @param seed The seed value.
+     */
     void SetRandEngineSeed(int seed);
+    /**
+     * @brief Sets the internal seed index for the random engine.
+     * @param index The seed index.
+     */
     void SetRandEngineInternalSeed(int index);
+    /**
+     * @brief Sets the random engine seed using system entropy.
+     */
     void SetRandSystemEntropySeed();
+    /**
+     * @brief Applies the random engine settings for the current thread.
+     * @return True if applied successfully.
+     */
     bool ApplyRandEngineForCurrentThread();
+    /**
+     * @brief Checks if the random engine is under controlled seeding.
```

Comment:
not true, if you read "controlled" as "user-supplied constant seed".
It will also be true when explicitely using system entropy.